diff --git a/plugins/pmaixforwardedfrom/pmaixforwardedfrom.c b/plugins/pmaixforwardedfrom/pmaixforwardedfrom.c
index 1163419..fa4a908 100644
--- a/plugins/pmaixforwardedfrom/pmaixforwardedfrom.c
+++ b/plugins/pmaixforwardedfrom/pmaixforwardedfrom.c
@@ -41,6 +41,7 @@
 #include "unicode-helper.h"
 
 MODULE_TYPE_PARSER
+MODULE_TYPE_NOKEEP
 PARSER_NAME("rsyslog.aixforwardedfrom")
 
 /* internal structures
