diff --git a/configure.ac b/configure.ac
index 331c77b..2930fa1 100644
--- a/configure.ac
+++ b/configure.ac
@@ -115,7 +115,7 @@ AC_TYPE_SIGNAL
 AC_FUNC_STAT
 AC_FUNC_STRERROR_R
 AC_FUNC_VPRINTF
-AC_CHECK_FUNCS([flock basename alarm clock_gettime gethostbyname gethostname gettimeofday localtime_r memset mkdir regcomp select setid socket strcasecmp strchr strdup strerror strndup strnlen strrchr strstr strtol strtoul uname ttyname_r getline malloc_trim prctl epoll_create epoll_create1 fdatasync lseek64])
+AC_CHECK_FUNCS([flock basename alarm clock_gettime gethostbyname gethostname gettimeofday localtime_r memset mkdir regcomp select setid socket strcasecmp strchr strdup strerror strndup strnlen strrchr strstr strtol strtoul sysinfo uname ttyname_r getline malloc_trim prctl epoll_create epoll_create1 fdatasync lseek64])
 
 # the check below is probably ugly. If someone knows how to do it in a better way, please
 # let me know! -- rgerhards, 2010-10-06
diff --git a/runtime/msg.c b/runtime/msg.c
index d874178..9f5c742 100644
--- a/runtime/msg.c
+++ b/runtime/msg.c
@@ -2875,8 +2875,8 @@ uchar *MsgGetProp(msg_t *pMsg, struct templateEntry *pTpe,
 			*pbMustBeFreed = 0;
 			break;
 		case PROP_SYS_UPTIME:
-#			ifdef OS_SOLARIS
-			pRes = (uchar*) "UPTIME NOT available under Solaris";
+#			ifndef HAVE_SYSINFO
+			pRes = (uchar*) "UPTIME NOT available on this system";
 			*pbMustBeFreed = 0;
 #			else
 			{
