Index: rsyslog/threads.c
===================================================================
--- rsyslog.orig/threads.c	2013-03-18 16:21:20.789144261 +0100
+++ rsyslog/threads.c	2013-03-21 16:31:05.424025586 +0100
@@ -183,10 +183,10 @@
 	assert(pThis != NULL);
 	assert(pThis->pUsrThrdMain != NULL);
 
+#	if HAVE_PRCTL && defined PR_SET_NAME
 	ustrncpy(thrdName+3, pThis->name, 20);
 	dbgOutputTID((char*)thrdName);
 
-#	if HAVE_PRCTL && defined PR_SET_NAME
 	/* set thread name - we ignore if the call fails, has no harsh consequences... */
 	if(prctl(PR_SET_NAME, thrdName, 0, 0, 0) != 0) {
 		DBGPRINTF("prctl failed, not setting thread name for '%s'\n", pThis->name);
