Index: syslogd.c
===================================================================
--- syslogd.c	(Revision 864)
+++ syslogd.c	(Arbeitskopie)
@@ -3675,6 +3675,7 @@
 	size_t iDirNameLen;
 	size_t iFileNameLen;
 	uchar szFullFileName[MAXFNAME];
+	uchar *ext = NULL;
 
 	assert(pDirName != NULL);
 
@@ -3696,6 +3697,10 @@
 			continue; /* we are not interested in special files */
 		if(res->d_name[0] == '.')
 			continue; /* these files we are also not interested in */
+		ext = strstr(res->d_name, ".conf");
+		if(ext == NULL || strlen(ext) != 5)
+			continue; /* Ignore files not ending in *.conf */
+
 		++iEntriesDone;
 		/* construct filename */
 		iFileNameLen = strnlen(res->d_name, NAME_MAX);
