Index: plugins/ommysql/createDB.sql
===================================================================
--- plugins/ommysql/createDB.sql	(Revision 994)
+++ plugins/ommysql/createDB.sql	(Arbeitskopie)
@@ -1,6 +1,4 @@
-CREATE DATABASE Syslog;
-USE Syslog;
-CREATE TABLE SystemEvents
+CREATE TABLE IF NOT EXISTS SystemEvents
 (
         ID int unsigned not null auto_increment primary key,
         CustomerID bigint,
@@ -28,7 +26,7 @@
         SystemID int NULL
 );
 
-CREATE TABLE SystemEventsProperties
+CREATE TABLE IF NOT EXISTS SystemEventsProperties
 (
         ID int unsigned not null auto_increment primary key,
         SystemEventID int NULL ,
Index: plugins/ompgsql/createDB.sql
===================================================================
--- plugins/ompgsql/createDB.sql	(Revision 994)
+++ plugins/ompgsql/createDB.sql	(Arbeitskopie)
@@ -1,5 +1,3 @@
-CREATE DATABASE "Syslog";
-\c Syslog;
 CREATE TABLE SystemEvents
 (
         ID serial not null primary key,
