Index: rsyslog/plugins/ommysql/createDB.sql
===================================================================
--- rsyslog.orig/plugins/ommysql/createDB.sql	2007-08-30 01:05:00.000000000 +0200
+++ rsyslog/plugins/ommysql/createDB.sql	2007-08-30 01:07:50.000000000 +0200
@@ -1,6 +1,4 @@
-CREATE DATABASE Syslog;
-USE Syslog;
-CREATE TABLE SystemEvents
+CREATE TABLE IF NOT EXISTS SystemEvents
 (
         ID int unsigned not null auto_increment primary key,
         CustomerID bigint,
@@ -28,7 +26,7 @@
         SystemID int NULL
 );
 
-CREATE TABLE SystemEventsProperties
+CREATE TABLE IF NOT EXISTS SystemEventsProperties
 (
         ID int unsigned not null auto_increment primary key,
         SystemEventID int NULL ,
